test/parallel/test-timers-setunreftimeout.js

```diff
@@ -0,0 +1,71 @@
+// Flags: --expose-internals
+
+'use strict';
+
+const common = require('../common');
+
+const { strictEqual } = require('assert');
+const { setUnrefTimeout } = require('internal/timers');
+
+// ERR_INVALID_CALLBACK
+{
+  // It throws ERR_INVALID_CALLBACK when first argument is not a function.
+  common.expectsError(
+    () => setUnrefTimeout(),
+    {
+      code: 'ERR_INVALID_CALLBACK',
+      type: TypeError,
+      message: 'Callback must be a function'
+    }
+  );
+}
+
+{
+  let called = false;
+  const timer = setUnrefTimeout(common.mustCall(() => {
+    called = true;
+  }), 1);
+
+  const testTimer = setUnrefTimeout(common.mustCall(() => {
+    strictEqual(called, false, 'unref pooled timer returned before check');
+    clearTimeout(testTimer);
+  }), 1);
+
+  strictEqual(timer.refresh(), timer);
+}
+
+// test case for passing the arguments
+{
+  for (let i = 3; i <= 6; i++) {
```

Comment:
If you run this with i = 0 ; i < 4; then you'll get the same results but will have to do less math.